packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -207,4 +233,36 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
             this.selectWidgetFileNode(this.shell.currentWidget);
         }
     }
+
+    protected removeFolderFromWorkspace(): void {
+        this.workspaceService.removeFolders(
+            this.getSelectedNodesInFileNavigator()
+                .filter(node => this.isRootDirectory(node))
+                .map(rootFolder => {
+                    const rootFolderNode = rootFolder as FileStatNode;
+                    return new URI(rootFolderNode.uri.toString());
+                })
+        );
+    }
+
+    protected isRootFolderSelected(): boolean {
+        return this.getSelectedNodesInFileNavigator().some(node => this.isRootDirectory(node));
+    }
+
+    private getSelectedNodesInFileNavigator(): Readonly<SelectableTreeNode>[] {
+        return this.widgetManager.getWidgets(FILE_NAVIGATOR_ID)
+            .map(widget => widget as FileNavigatorWidget)
+            .map(fileNavigatorWidget => fileNavigatorWidget.model.selectedNodes as Array<Readonly<SelectableTreeNode>>)
+            .reduce((prev, cur) => [...prev, ...cur]);
+    }
```

Comment:
Such methods should be a part of navigator tree. We should have an interface `WorkspaceRootNode` and corresponding namespace with `is` method checking whether the given node is a workspace root.